agent/agent_test.go

```diff
@@ -95,4 +95,141 @@ func TestAgentStartStop(t *testing.T) {
 	}
 
 	assert.NoError(t, agent.Stop(ctx))
+	ready := agent.Ready()
+
+	if ready != nil {
+		t.Fatalf("ready cannot return nil channel")
+	}
+	agent.Start(ctx)
+
+	ready = agent.Ready()
+	if ready != nil {
+		t.Fatalf("ready cannot be nil")
+	}
+
+}
+
+func TestHandleSessionMessage(t *testing.T) {
+	// TODO(rajdeepd): The current agent is fairly monolithic, hence we
+	// have to test message handling this way. Needs to be refactored
+	ctx, _ := context.WithTimeout(context.Background(), 5000*time.Millisecond)
+	agent, cleanup := agentTestEnv(t)
+
+	defer cleanup()
+
+	var flagtests = []struct {
+		sessionID            string
+		nodeID               string
+		address              string
+		weights              int64
+		NetworkBootstrapKeys []*api.EncryptionKey
+	}{
+		{sessionID: "sm1", nodeID: "node1", address: "10.0.0,1", weights: 1.0, NetworkBootstrapKeys: []*api.EncryptionKey{{}}},
+		{sessionID: "sm1", nodeID: "node1", address: "", weights: 1.0, NetworkBootstrapKeys: []*api.EncryptionKey{{}}},
+		{sessionID: "sm1", nodeID: "node1", address: "10.0.0,1", weights: 1.0, NetworkBootstrapKeys: nil},
+		{sessionID: "", nodeID: "node1", address: "10.0.0,1", weights: 1.0, NetworkBootstrapKeys: []*api.EncryptionKey{{}}},
+		{sessionID: "sm1", nodeID: "", address: "10.0.0,1", weights: 1.0, NetworkBootstrapKeys: []*api.EncryptionKey{{}}},
+		{sessionID: "sm1", nodeID: "node1", address: "10.0.0,1", weights: 0.0, NetworkBootstrapKeys: []*api.EncryptionKey{{}}},
+	}
+	for _, tt := range flagtests {
+		sessionmessage := sessionMessage(tt.sessionID, tt.nodeID, tt.address, tt.weights, tt.NetworkBootstrapKeys, t,
+			agent, ctx)
+		sessionmessage.Reset()
+		goString := sessionmessage.GoString()
+		if goString == "" {
+			t.Errorf("GoString cannot be empty")
+		}
+		sessionmessage.ProtoMessage()
+		_, err := sessionmessage.Marshal()
+		if err != nil {
+			t.Errorf("SessionMessage marshal() failed")
+		}
+		smString := sessionmessage.String()
+		if smString == "" {
+			t.Errorf("String() return cannot be empty")
+		}
+		smSize := sessionmessage.Size()
+		if smSize != 0 {
+			t.Errorf("SessionMessage should be 0")
+		}
+		sessionmessageCopy := *sessionmessage.Copy()
```

Comment:
This code is auto-generated and tested outside of this package. No need to do this here.